cmd/kube-apiserver/app/server.go

```diff
@@ -181,8 +182,12 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.CertDirectory, "cert-dir", s.CertDirectory, "The directory where the TLS certs are located (by default /var/run/kubernetes). "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
 	fs.StringVar(&s.APIPrefix, "api-prefix", s.APIPrefix, "The prefix for API requests on the server. Default '/api'.")
-	fs.MarkDeprecated("api-prefix", "--api-prefix is deprecated and will be removed when the v1 API is retired")
-	fs.StringVar(&s.StorageVersion, "storage-version", s.StorageVersion, "The version to store resources with. Defaults to server preferred")
+	fs.MarkDeprecated("api-prefix", "--api-prefix is deprecated and will be removed when the v1 API is retired.")
+	fs.StringVar(&s.StorageVersion, "storage-version", s.StorageVersion, "The version to store the legacy v1 resources with. Defaults to server preferred")
+	fs.MarkDeprecated("storage-version", "--storage-version is deprecated and will be removed when the v1 API is retired. See --storage-versions instead.")
+	fs.StringVar(&s.StorageVersions, "storage-versions", s.StorageVersions, "The versions to store resources with."+
+		"Different groups may be stored in different versions. Specified in the format \"group1/version1,group2/version2...\"."+
+		"Defaults to server preferred.")
```

Comment:
Let's talk about this default-- I'm kind of inclined to argue that we want to make admins explicitly set this, because changes in this imply the necessity of running the upgrade script. @smarterclayton - do you or any other redhatter have an opinion on this?